### ISSUE TITLE
Cmake POSITION_INDEPENDENT_CODE support

### ISSUE THREAD

**Issue by aTom3333** [illuhad/hipSYCL#296]
Adding support for cmake's `POSITION_INDEPENDENT_CODE` property for syclcc would be nice.
As of now, with the following setup I get a linker error.

CMakeLists.txt:
```cmake_minimum_required(VERSION 3.16)
project(hipsycl_pic)

set(HIPSYCL_PLATFORM cpu)
find_package(hipsycl CONFIG PATHS /path/to/hipSYCL/install/lib/cmake/)

add_library(hipsycl_pic_lib SHARED add.h add.cpp)
add_sycl_to_target(TARGET hipsycl_pic_lib SOURCES add.cpp)

set_target_properties(hipsycl_pic_lib PROPERTIES
    POSITION_INDEPENDENT_CODE ON
)

include(GenerateExportHeader)
generate_export_header(hipsycl_pic_lib
        BASE_NAME mylib
        )
target_include_directories(hipsycl_pic_lib PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
```

add.h:
```#ifndef ADD_H
#define ADD_H

#include "mylib_export.h"

#ifdef __cplusplus
extern "C" {
#endif

void MYLIB_EXPORT add(float* a, float* b, float* out, unsigned int n);
    
#ifdef __cplusplus
};
#endif
#endif // ADD_H
```

add.cpp:
```#include "add.h"
#include <CL/sycl.hpp>

using namespace cl::sycl;
void add(float* a, float* b, float* out, unsigned int n)
{
    static queue q;
    buffer<float> aBuf(a, n);
    buffer<float> bBuf(b, n);
    buffer<float> outBuf(out, n);
    q.submit([&](handler& h){
        auto aAcc = aBuf.get_access<access::mode::read>(h);
        auto bAcc = bBuf.get_access<access::mode::read>(h);
        auto outAcc = outBuf.get_access<access::mode::write>(h);
        h.parallel_for<class sycl_kernel>(range<1>(n), [=](id<1> id) {
            outAcc[id] = aAcc[id] + bAcc[id];
        });
    });
}
```

Resulting linker error:
```/usr/bin/ld: CMakeFiles/hipsycl_pic_lib.dir/add.cpp.o: warning: relocation against '_ZNSt17_Function_handlerIFvvEZN6hipcpu6detail11async_queue4workEvEUlvE0_E10_M_managerERSt9_Any_dataRKS6_St18_Manager_operation' in read-only section '.text._ZNSt8functionIFvvEEC2IZN6hipcpu6detail11async_queue4workEvEUlvE0_vvEET_[_ZNSt8functionIFvvEEC5IZN6hipcpu6detail11async_queue4workEvEUlvE0_vvEET_]'
/usr/bin/ld: CMakeFiles/hipsycl_pic_lib.dir/add.cpp.o: relocation R_X86_64_PC32 against symbol '_ZN7hipsycl4sycl5queueD1Ev' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
```

As the linker says, the problem is that the `-fPIC` isn't given to the compiler when compiling for the cpu.

For my use case, i changed syclcc-clang to add the flag unconditionnaly in the `pure_cpu_compiler` class but it would be better if the syclcc toolchain could understand the `POSITION_INDEPENDENT_CODE` property and add the flag accordingly.

**Comment by illuhad** [comment]
I don't think this is an issue with `syclcc` - it just forwards any of its arguments that are not hipSYCL-specific to the underlying compiler, so `syclcc -fPIC` will work. The issue is probably rather that the hipSYCL cmake integration does not implement the `POSITION_INDEPENDENT_CODE` feature.

**Comment by aTom3333** [comment]
That's what i think too, excuse me if I wasn't clear, when i said toolchain, i meant toolchain as in the cmake sense.
If I had to guess, I'd say the CMakeSYCLInformation.cmake file should be changed

**Comment by illuhad** [comment]
You may be right with your guess. I'm not sure how we would need to implement this in cmake. Would you be interested in investigating and maybe creating a PR?

**Comment by aTom3333** [comment]
fixed by #299
